### ISSUE TITLE
Stored queries in the sidebar render as 'selected'

### ISSUE THREAD

**Issue by mechanarchy** [beancount/fava#1358]
When queries are saved in the beancount file, these are rendered in the Fava sidebar as clickable links. However, navigating to their parent `Query` item causes these children to always appear as 'selected'. This prevents the hover animation and the overly-dark background colour is out of place with the rest of the sidebar:

![image](https://user-images.githubusercontent.com/1166756/149124041-293ee4a8-dd8b-46fd-b066-4c62d07009e6.png)

I tested this patch and it turns off the permanent 'selected' appearance for children item now. A side effect of this is that, while the hover-highlighting is now functional, clicking one of these child queries does not turn on the 'selected' state. I think this is appropriate as the child items are not different pages which are being navigated to; but really shortcuts to automatically execute a given query on the page.

```patch
diff --git a/frontend/src/sidebar/index.ts b/frontend/src/sidebar/index.ts
index 7ecaa8d2..b83cc72a 100644
--- a/frontend/src/sidebar/index.ts
+++ b/frontend/src/sidebar/index.ts
@@ -11,7 +11,8 @@ export function updateSidebar(): void {
     const href = el.getAttribute("href");
     if (
       !el.hasAttribute("data-remote") &&
-      href?.includes(window.location.pathname)
+      href?.includes(window.location.pathname) &&
+      !el.parentElement?.parentElement?.classList?.contains("submenu")
     ) {
       el.classList.add("selected");
     }
```

This issue potentially relates to #396 , ab3785e

**Comment by yagebu** [comment]
Thanks for the report and the suggested fix :) Fixed in a slightly modified way in 4e541fbc42d69e81a1f3388060355b80cd7b8226

**Comment by mechanarchy** [comment]
Thanks! I though my fix had room for improvement, hence not making it a PR at the start.
